### ISSUE TITLE
How to connect secured elasticsearch cluster?

### ISSUE THREAD

**Issue by joshuaavalon** [FusionAuth/fusionauth-issues#531]
### Description

I want to try out FusionAuth by connecting to a cloud elasticsearch cluster that secured with authentication. However, I cannot find the documents on how to configurate it.

**Comment by robotdan** [comment]
What service are you wanting to connect to and what authentication methods for they support?

**Comment by joshuaavalon** [comment]
I can connect it through http://user:password@example.com.

However, I have problem on give permissions to the application. Why permissions does this application need?

**Comment by nscarlson** [comment]
@robotdan I'm having the same issue attempting to connect FusionAuth with an external, hosted elasticsearch cluster.  It seems that FusionAuth doesn't like elasticsearch URLs containing basic auth credentials, which are the de facto standard for hosted ES providers.

If `FUSIONAUTH_SEARCH_SERVERS` is not defined, on startup FusionAuth starts with maintenance mode requesting an elasticsearch URL.  If provided with an elasticsearch URL with basic auth credentials, such as `http://user:password@example.com`, and error will display saying something that the provided URL is not elasticsearch.

If `FUSIONAUTH_SEARCH_SERVERS` is defined with a URL containing basic auth credentials, then on startup the generic silent configuration failed screen is shown.

Can corroborate with more info/screenshots later, but it's late :)

**Comment by robotdan** [comment]
@nscarlson or @joshuaavalon do you have any example services that still use this style of authentication? 

I have updated our REST client to use basic auth using the the pattern `http://user:password@example.com`. I can confirm I can use this style of URL and I can connect to different services. But I haven't found one to test the authentication works. I've mainly used vanilla, and AWS. 

We're essentially parsing the URL , extracting the credentials and then using the the Basic Credentials Provider in the Elasticsearch client builder. 

https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_basic_authentication.html

**Comment by nscarlson** [comment]
@robotdan Currently I'm using https://bonsai.io.  Their basic auth credentials are of the form:

```
https://user:password@domain.com:443
```

<img width="579" alt="Screen Shot 2020-05-07 at 7 00 01 PM" src="https://user-images.githubusercontent.com/3528679/81358921-0f555980-9095-11ea-911a-9e60d83619ad.png">

Interestingly if I omit the trailing `:443` I get the Tomcat 404:

<img width="1051" alt="Screen Shot 2020-05-07 at 7 02 07 PM" src="https://user-images.githubusercontent.com/3528679/81359057-56434f00-9095-11ea-895b-ab4346c6e119.png">

**Comment by robotdan** [comment]
Ok, perfect, thanks for the additional info. 

I had just signed up for a trial at https://aiven.io/ and tested theirs since they also use that authentication style. Worked, so I think we're good to go. 

I just tried https://app.bonsai.io/ and that worked as well. I tried removing `:443` on the URL and it still worked fine.  I re-worked how we parse that URL to build the Elasticsearch REST client, so it seems to be working better. 

This should go out in 1.16.0, and you'll have to let me know how it goes.
